4-WebApp-your-API/4-1-MyOrg/README-use-certificate.md

```diff
@@ -0,0 +1,95 @@
+# Web app using client certificates with Microsoft.Identity.Web
+
+There are different approaches to [use certificates with Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web/wiki/Certificates). In this Readme, we will explore use of certificate from a path on the disk.
+
+## Use Self Signed Certificate from a path
+
+To use certificates instead of an application secret you will need to do some changes to what you have done so far:
+
+- generate a certificate and export it, if you don't have one already
+- register the certificate with your application in the Azure AD portal
+- Update appsettings.json
+
+### Generate a Self Signed certificate
+
+#### with Powershell
+
+To complete this step, you will use the `New-SelfSignedCertificate` Powershell command. You can find more information about the New-SelfSignedCertificate command [here](https://docs.microsoft.com/powershell/module/pkiclient/new-selfsignedcertificate).
+
+1. Open PowerShell and run New-SelfSignedCertificate with the following parameters to create a self-signed certificate in the user certificate store on your computer:
+
+    ```PowerShell
+    $cert=New-SelfSignedCertificate -Subject "CN=WebAppCert" -CertStoreLocation "Cert:\CurrentUser\My"  -KeyExportPolicy Exportable -KeySpec Signature
+    ```
+
+1. Export this certificate using the "Manage User Certificate" MMC snap-in accessible from the Windows Control Panel. You can also add other options to generate the certificate in a different
+store such as the Computer or service store (See [How to: View Certificates with the MMC Snap-in](https://docs.microsoft.com/dotnet/framework/wcf/feature-details/how-to-view-certificates-with-the-mmc-snap-in)).
+
+Alternatively you can use an existing certificate if you have one (just be sure to record its name for the next steps).
+
+Export one with private key as WebAppCert.pfx and another as WebAppCert.cer without private key.
+
+#### with OpenSSL
+
+Type the following in a terminal. You will be prompted to enter some identifying information and a *passphrase*:
+
+```console
+    openssl req -x509 -newkey rsa:4096 -sha256 -days 365 -keyout webapp.key -out webapp.cer -subj "/CN=webapp.com" -addext "subjectAltName=DNS:webapp.com"
```

Comment:
```suggestion
    openssl req -x509 -newkey rsa:4096 -sha256 -days 365 -keyout webapp.key -out webapp.cer -nodes -batch
```
putting this here will do the following:
-nodes (means `no DES`) : no need for a passphrase for DES encryption
-batch : don't prompt for any attributes like common name, dns, etc. This cert doesn't need these.